### PR TITLE
Fixed: Parsing of quality when release group contains Remux

### LINKED ISSUE #6607: Pull Sonarr commit 'Fixed: Parsing of quality when release group contains Remux'
Pull https://github.com/Sonarr/Sonarr/commit/8ad5e5dd13a43b69fcbb8762489ee882745e1e2f

```
git cherry-pick -ex 8ad5e5dd13a43b69fcbb8762489ee882745e1e2f
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

[AB#1623](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1623)

### PR DESCRIPTION
**Pull request by austinwbest**
* Fixes #6607

### CODE REVIEW

**Comment by austinwbest** [comment]
I'll look at the regex, seems it works for the changed 720 tests (the only one i tested locally) but not some 1080's

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ public class QualityParser
 
         private static readonly Regex HighDefPdtvRegex = new Regex(@"hr[-_. ]ws", RegexOptions.Compiled | RegexOptions.IgnoreCase);
 
-        private static readonly Regex RemuxRegex = new Regex(@"\b(?<remux>(BD|UHD)?[-_. ]?Remux)\b", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static readonly Regex RemuxRegex = new Regex(@"[_. ](?<remux>(?:(BD|UHD)[-_. ])?Remux)\b", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Parser/QualityParser.cs:76]:
```suggestion
        private static readonly Regex RemuxRegex = new Regex(@"[_. ](?<remux>(?:(BD|UHD)[-_. ]?)?Remux)\b", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
should fix the testcase that failed

**Review by bakerboy448 — DISMISSED** [review]

**Comment by bakerboy448** [comment]
I see no reason why the REMUX unrelated quality tests should be failing; the regex is the same

image
image

File: src/NzbDrone.Core/Parser/QualityParser.cs [hunk 2]
```diff
@@ -73,7 +73,7 @@ public class QualityParser
 
         private static readonly Regex HighDefPdtvRegex = new Regex(@"hr[-_. ]ws", RegexOptions.Compiled | RegexOptions.IgnoreCase);
 
-        private static readonly Regex RemuxRegex = new Regex(@"\b(?<remux>(BD|UHD)?[-_. ]?Remux)\b", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static readonly Regex RemuxRegex = new Regex(@"(?:[_. ]|\d{4}p-)(?<remux>(?:(BD|UHD)[-_. ]?)?Remux)\b", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Parser/QualityParser.cs:76]: allows for 1080p-Remux and 2160p-Remux

**Comment by bakerboy448** [comment]
Test was failing due to `1080p-Remux` with `-` 

reworked code to allow for {4 digits}p-Remux

note: sonarr was missing these remux tests - PR'd https://github.com/Sonarr/Sonarr/pull/4740

**Review by Qstick — APPROVED** [review]

**Pull request merged**